Ticket 3787 - Add ioc aldn1000

aaron-long opened the pull request:
### Description of work

Created OPI for the WPI Aladdin 1000 syringe pump.

### Ticket

https://github.com/ISISComputingGroup/IBEX/issues/3787

### Acceptance criteria

- [x] I can control the WPI Aladdin-1000 Syringe Pump via the IBEX GUI

### Documentation

https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/WPI-Aladdin-Syringe-Pump

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?
- [ ] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?

### Final Steps
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

DominicOram commented:
Couple of bits:
* Your message about rate units is a bit confusing, maybe instead "Rate units will not change on the device  until a new pumping rate has been set"
Although not specifically UI problems these are obvious when running the GUI against a devsim device:
* Pump status and Last set error are both in error
* The clear volume errors when set (and there's no way to test this as the emulator doesn't actually do any pumping)
* Pump address, does the user regularly talk to lots of different pumps? What is the behaviour on the device when you switch address? Do the readbacks all change? Is this behaviour emulated correctly?
* Does the program function change at all? If not why does the user need to see it?

aaron-long commented:
Bits:
- [x] Updated rate units label for clarity
- [x] Fixed
- [x] Added WPI entry into IGNORED_WORDS list

Additional bits:
- [x] Fixed error states for status/last error
- [x] Moved pump and program function into an advanced tab. This should help reduce confusion as discussed
